csharp/src/Apache.Arrow.Flight/FlightClient.cs

```diff
@@ -0,0 +1,90 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Apache.Arrow.Flight.Protocol;
+using Apache.Arrow.Flight.Writer;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace Apache.Arrow.Flight
+{
+    public class FlightClient
+    {
+        private readonly FlightService.FlightServiceClient _client;
+        public FlightClient(GrpcChannel grpcChannel)
+        {
+            _client = new FlightService.FlightServiceClient(grpcChannel);
+        }
+
+        public IAsyncStreamReader<FlightInfo> ListFlights(Criteria criteria = null, Metadata headers = null)
+        {
+            if(criteria == null)
+            {
+                criteria = new Criteria();
+            }
+            
+            var response = _client.ListFlights(criteria.ToProtocol(), headers);
+            return new StreamReader<Protocol.FlightInfo, FlightInfo>(response.ResponseStream, inFlight => new FlightInfo(inFlight));
+        }
+
+        public IAsyncStreamReader<ActionType> ListActions(Metadata headers = null)
+        {
+            var response = _client.ListActions(new Empty(), headers);
```

Comment:
Similar comment for `new Empty()`. Instead make an `internal static readonly Empty Instance = new Empty()` field and use it here.